src/grape_block.rs

```diff
@@ -0,0 +1,143 @@
+//! Implements the GrapeBlock, a monitor wrapper for each CyberGrape
+//! processing module. Enforces a common interface between module such that
+//! each module can consume data from the preceding module, process it, and
+//! pass new data to the subsequent module in the CyberGrape pipeline.
+
+use crate::Component;
+use log::{info, warn};
+use std::fmt;
+use std::sync::mpsc::{channel, Receiver, Sender};
+
+///
+/// Runs on its own thread. On receiving data of type A on input, the
+/// GrapeBlock converts them to data of type B, and sends it to its output
+/// channel.
+///
+pub struct GrapeBlock<'a, A, B> {
+    // component is an object of trait Component, and Rust cannot know
+    // how big component is at runtime. any 'dynamically typed' trait
+    // object must be marked with the `dyn` keyboard, post 2021
+
+    // the Box ensures that this GrapeBlock owns the heap-alloc'd ref to the
+    // underlying Component
+
+    // the lifetime 'a indicates that the inner Component cannot live longer
+    // than the GrapeBlock!
+    component: Box<dyn Component<InData = A, OutData = B> + 'a>,
+    output: Sender<B>,
+    input: Receiver<A>,
+}
```

Comment:
I think that we should stick to `I` and `O` as type variables like in the `impl` below.